src/buffer/out/textBuffer.cpp

```diff
@@ -572,8 +641,7 @@ bool TextBuffer::IncrementCircularBuffer(const bool inVtMode)
         // the current background color, but with no meta attributes set.
         fillAttributes.SetStandardErase();
     }
-    const auto fSuccess = _storage.at(_firstRow).Reset(fillAttributes);
-    if (fSuccess)
+    GetRowByOffset(0).Reset(fillAttributes);

```

Comment:
wait... there's a comment on `_firstRow` saying that it's not necessarily 0. So shoult this actually be...
```cpp
    GetRowByOffset(_firstRow).Reset(fillAttributes);
```

Reply:
`GetRowByOffset` returns `_storage[_firstRow + index]`. Because of that, `_storage.at(_firstRow)` and `GetRowByOffset(0)` are equivalent. I changed this code as I was adding a lazy initialization of rows, which only works if everyone uses `GetRowByOffset` - even internal callers like this one. But I removed it to keep this first PR more compact. I'll be adding it in a future PR.

Reply:
Do we even still have the row cycling system? Are you intending on replacing it with buffer recycling so that we don't need to reorder rows? Or is it fine as it is?

Reply:
`row cycling`? If you mean whether we're still using a circular buffer, then yeah, we do that. I'm just intending to move the `ROW::_init()` call from the constructor over to `GetRowByOffset()` so that row initialization happens lazily.